src/components/FilterSearch.tsx

```diff
@@ -119,25 +125,35 @@ export function FilterSearch({
     const newDisplayName = itemData?.displayName as string;
     if (newFilter && newDisplayName) {
       if (select) {
-        if (currentFilter) {
-          searchActions.setFilterOption({ filter: currentFilter, selected: false });
-        }
-        searchActions.setFilterOption({ filter: newFilter, displayName: newDisplayName, selected: true
-        });
-        setCurrentFilter(newFilter);
-        setFilterQuery(newDisplayName);
-        executeFilterSearch(newDisplayName);
-        if (searchOnSelect) {
-          searchActions.setOffset(0);
-          searchActions.resetFacets();
-          executeSearch(searchActions);
+        if (onSelect && !searchOnSelect) {
```

Comment:
This is getting a little messy, can we move to it's own function `handleSelect` and use gates like

```
if (!newFilter && !newDisplayName) {
  return;
}

if (!select) {
  return;
}

if (onSelect && !searchOnSelect) {
  onSelect(...)
} else {
  ...
}
```

Reply:
I also think some refactoring here would be good since we're adding additional complexity in this PR!

Reply:
Did some refactoring to simplify things. Turns out, the `handleToggleDropdown` code was no longer needed because we're not doing special behavior when a user clicks out of filter search